compilerlibs/Makefile.compilerlibs

```diff
@@ -430,8 +432,15 @@ compilerlibs/ocamlcommon.cma: $(COMMON_CMI) $(ALL_CONFIG_CMO) $(COMMON)
 partialclean::
 	rm -f compilerlibs/ocamlcommon.cma
 
-compilerlibs/ocamlcommon.cmxa: $(COMMON_CMI) $(COMMON:.cmo=.cmx)
-	$(V_LINKOPT)$(CAMLOPT) -a -linkall -o $@ $(COMMON:.cmo=.cmx)
+ZSTD_OBJ=runtime/zstd.n.$(O)
+ZSTD_FLAGS=$(patsubst %, -ccopt %, $(filter-out -l%,$(ZSTD_LIBS))) \
+           $(patsubst %, -cclib %, $(filter -l%,$(ZSTD_LIBS)))
+
+compilerlibs/ocamlcommon.cmxa: $(COMMON_CMI) $(COMMON:.cmo=.cmx) $(ZSTD_OBJ)
+	$(V_LINKOPT)$(CAMLOPT) -a -linkall -o $@ \
+                               $(ZSTD_FLAGS) $(COMMON:.cmo=.cmx)
+	$(V_MKLIB)$(ARCMD) r compilerlibs/ocamlcommon.$(A) $(ZSTD_OBJ)
```

Comment:
This creative use of `ar` is due to @dra27.  If we just add `zstd.n.o` to the `ocamlopt -a` command line, this file name is recorded in `ocamlcommon.cmxa` and passed to the system linker when the latter file is used, but the system linker will not find it and will not search for it in the `-L` search path.  Instead, we include `zstd.n.o` and its contents in the `ocamlcommon.a` archive, where the system linker will find it reliably.

It's unclear whether this hack can work with the MSVC librarian `lib`.  Good thing we're not supporting MSVC again yet.

To make things cleaner in the future, we can either
- Change `ocamlopt -a` so that all `.o` files given on the command line are included in the generated `.a` file.  The current behavior (of recording the file names in the `.cmxa` file) is useless anyway.
- Put `zstd.n.o` in a proper `.a` archive, and install this archive.  This would be the case if we make an `otherlibs` that exposes compressed marshaling to users.
-